runtime/src/proofs.rs

```diff
@@ -66,12 +90,29 @@ fn validate_proof(matches: &mut Vec<[u8; 32]>, hash: [u8; 32], proofs: Vec<[u8;
     false
 }
 
+// prefill the required matches before proof validation
+fn pre_matches(static_proofs: [[u8; 32]; 3], doc_root: [u8; 32]) -> (bool, Vec<[u8; 32]>) {
```

Comment:
Is the `static_proofs: [[u8; 32]; 3]` length check implicit here, or do we need to add an extra check on required length of 3, to avoid out of bounds exception?

Reply:
its implicit here. It should only have 3 and not more
If any missing, then the hash check fails

Reply:
Cool, because it is initialized to 0s to the ones that are not provided, right?